JSIL.Libraries/Sources/JSIL.Core.js

```diff
@@ -4546,81 +4581,119 @@ JSIL.InitializeType = function (type) {
   else
     return;
 
-  if (typeObject.__TypeInitialized__ || false)
+  if (typeObject.__TypeInitialized__ || typeObject.__TypeInitializing__ || false)
     return;
 
-  // Not entirely correct, but prevents recursive type initialization
-  typeObject.__TypeInitialized__ = true;
+  var shouldDoScheduledInitialization = !JSIL.IsInsideInitializeType;
+
+  try {
+    var baseType = typeObject.__BaseType__;
+    while (baseType || false) {
+      if (baseType.__TypeInitializing__) {
+        if (JSIL.ScheduledInitialization.indexOf(typeObject) == -1) {
+          JSIL.ScheduledInitialization.push(typeObject);
+        }
+        return;
+      }
+      baseType = baseType.__BaseType__;
+    }
+
+    JSIL.IsInsideInitializeType = true;
+    typeObject.__TypeInitializing__ = true;
 
-  if (typeObject.__IsClosed__) {
-    var forceLazyMethodGroups = false;
+    if (typeObject.__IsClosed__) {
```

Comment:
try { was added to line 4589, so indentation was increased. Yes, diff in whitespace.